core/src/test/java/org/apache/calcite/test/RexProgramTest.java

```diff
@@ -1733,6 +1724,20 @@ private void checkExponentialCnf(int n) {
         "true");
   }
 
+  @Test public void testSimplifyItemRangeTerms() {
+    RexNode item = item(input(tArray(tInt()), 3), literal(1));
+    // paranoid validation doesn't support array types, disable it for a moment
+    simplify = this.simplify.withParanoid(false);
+    // (a=1 or a=2 or (arr[1]>4 and arr[1]<3 and a=3)) => a=1 or a=2
+    checkSimplifyFilter(
+        or(
+          eq(vInt(), literal(1)),
+          eq(vInt(), literal(2)),
+          and(gt(item, literal(4)), lt(item, literal(3)), eq(vInt(), literal(3)))),
+        "OR(=(?0.int0, 1), =(?0.int0, 2))");
+    simplify = simplify.withParanoid(true);
```

Comment:
This could probably be removed since JUnit does not reuse test instances by default.